src/goto-instrument/contracts/contracts.cpp

```diff
@@ -271,24 +288,29 @@ void code_contractst::check_apply_loop_contracts(
   }
   else
   {
-    assigns.insert(
+    // An assigns clause was specified for this loop.
+    // Add the targets to the set of expressions to havoc.
+    // TODO: Should we add the automatically detected local static variables
+    // too ? (they are present in loop_assigns but not in assigns_clause, and
+    // they are not necessarily touched by the loop).
```

Comment:
Who might be able to answer that question, or how would we find out the answer?

Reply:
We should track this in a Github issue as well.

Reply:
This needs careful analysis, keeping it here.